packages/flutter_tools/lib/src/migrations/swift_package_manager_integration_migration.dart

```diff
@@ -0,0 +1,1083 @@
+  static const String _localFlutterPluginsSwiftPackageReferenceIdentifer = '781AD8BC2B33823900A9FFBB';
+
+  /// New identifer for FlutterGeneratedPluginSwiftPackage XCSwiftPackageProductDependency.
+  static const String _flutterPluginsSwiftPackageProductDependencyIdentifer = '78A3181F2AECB46A00862997';
+
+  /// Existing iOS identifer for Runner PBXFrameworksBuildPhase.
+  static const String _iosRunnerFrameworksBuildPhaseIdentifer = '97C146EB1CF9000F007C117D';
+
+  /// Existing macOS identifer for Runner PBXFrameworksBuildPhase.
+  static const String _macosRunnerFrameworksBuildPhaseIdentifer = '33CC10EA2044A3C60003C045';
+
+  /// Existing iOS identifer for Runner PBXNativeTarget.
+  static const String _iosRunnerNativeTargetIdentifer = '97C146ED1CF9000F007C117D';
+
+  /// Existing macOS identifer for Runner PBXNativeTarget.
+  static const String _macosRunnerNativeTargetIdentifer = '33CC10EC2044A3C60003C045';
+
+  /// Existing iOS identifer for Runner PBXProject.
+  static const String _iosProjectIdentifier = '97C146E61CF9000F007C117D';
+
+  /// Existing macOS identifer for Runner PBXProject.
+  static const String _macosProjectIdentifier = '33CC10E52044A3C60003C045';
+
+  /// Path to add to .gitignore so generated files in FlutterGeneratedPluginSwiftPackage
+  /// are not committed.
+  static const String flutterPackageGitignore = '**/Flutter/Packages/FlutterGeneratedPluginSwiftPackage';
+
+  File get backupProjectSettings => _fileSystem
+      .directory(_xcodeProjectInfoFile.parent)
+      .childFile('project.pbxproj.backup');
+
+  String get _runnerFrameworksBuildPhaseIdentifer {
+    return _platform == SupportedPlatform.ios
+        ? _iosRunnerFrameworksBuildPhaseIdentifer
+        : _macosRunnerFrameworksBuildPhaseIdentifer;
+  }
+
+  String get _runnerNativeTargetIdentifer {
+    return _platform == SupportedPlatform.ios
+        ? _iosRunnerNativeTargetIdentifer
+        : _macosRunnerNativeTargetIdentifer;
+  }
+
+  String get _projectIdentifier {
+    return _platform == SupportedPlatform.ios
+        ? _iosProjectIdentifier
+        : _macosProjectIdentifier;
+  }
+
+  void restoreFromBackup(SchemeInfo? schemeInfo) {
+    if (backupProjectSettings.existsSync()) {
+      _logger.printTrace('Restoring project settings from backup file...');
+      backupProjectSettings.copySync(_xcodeProject.xcodeProjectInfoFile.path);
+    }
+    schemeInfo?.backupSchemeFile?.copySync(schemeInfo.schemeFile.path);
+  }
+
+  /// Add Swift Package Manager integration to Xcode project's project.pbxproj
+  /// and Runner.xcscheme.
+  ///
+  /// If migration fails or project.pbxproj or Runner.xcscheme becomes invalid,
+  /// will revert any changes made and throw an error.
+  ///
+  /// Also, adds [flutterPackageGitignore] to .gitignore if found.
+  Future<void> migrate() async {
+    Status? migrationStatus;
+    SchemeInfo? schemeInfo;
+    try {
+      if (!_xcodeProjectInfoFile.existsSync()) {
+        throw Exception('Xcode project not found.');
```

Comment:
```suggestion
        throwToolExit('Xcode project not found.');
```

Reply:
Does it matter if it's `throwToolExit` since this is in a try catch where the catch will rethrow as a `throwToolExit`?

Reply:
Ah, missed that.  I think what you have is fine then?

Reply:
Yeah, it's fine. I just remembered my unit tests expect `throwsToolExit` so it confirms it's throwing the correct thing